src/js/messaging/reducers/compose.js

```diff
@@ -32,17 +53,19 @@ function getRecipients(recipients) {
 
 export default function compose(state = initialState, action) {
   switch (action.type) {
-    case SET_CATEGORY:
-      return set('category', action.field.value, state);
-    case SET_SUBJECT:
-      return set('subject.value', action.field.value, state);
+    case SET_MESSAGE:
+      return set(action.path, action.field.value, state);
     case SET_SUBJECT_REQUIRED:
-      return set('subject.required', action.fieldState.required, state);
-    case SET_RECIPIENT:
-      return set('recipient', action.field.value, state);
+      return set('message.subject.required', action.fieldState.required, state);
     case FETCH_RECIPIENTS_SUCCESS:
       return set('recipients', getRecipients(action.recipients.data), state);
+    case FETCH_SENDER_SUCCESS:
+      return set('message.sender', action.sender, state);
+    case SHOW_CONFIRM_DELETE:
+      return set('modals.deleteconfirm.visible', !state.modals.deleteconfirm.visible, state);
```

Comment:
If the one action switches the dialog on and off, I would call it something like `TOGGLE_CONFIRM_DELETE` rather than just `SHOW_*`.